### PR TITLE
Joseph lunt

### PR DESCRIPTION
**Pull request by GoJoeyGo**

### CODE REVIEW

**Review by MinaTamang — COMMENTED** [review]
c76a17538f3b6a4a0aa94a2505db615b40b6b6cc
Joey can you change boolean Damaged into boolean isDamaged. I think that will look appropriate.

**Review by MinaTamang — CHANGES_REQUESTED** [review]
@GoJoeyGo  can you change line 32 of Loan.java. getOverDueLoan into boolean isLoanOverDue. I think that will be more appropriate.
Thanks.

**Comment by MinaTamang** [comment]
@GoJoeyGo ignore the changes requested message

**Review by MinaTamang — COMMENTED** [review]
@GoJoeyGo  can you change line 32 of Loan.java. getOverDueLoan into boolean isLoanOverDue. I think that will be more appropriate.
Thanks.

**Review by MinaTamang — COMMENTED** [review]
@GoJoeyGo can you update the line 16 and 17 private List<book> into private List<loan> into private List<Book> and private List<Loan>. I think that's little right way.
And line 53 scanneBook into scanBook or something its just spelling error. Thankyou

**Review by MinaTamang — COMMENTED** [review]
@GoJoeyGo can you please change line 9 private scanner keybordinput into keyboardInput aand line 112 public void display into Display().
Thanks

File: ReturnBookUI.java [hunk 1]
```diff
@@ -3,40 +3,41 @@
...
 public class ReturnBookUI {
 
-	public static enum UI_STATE { INITIALISED, READY, INSPECTING, COMPLETED };
+	public static enum ReturnBookUiState { INITIALISED, READY, INSPECTING, COMPLETED };
 
-	private ReturnBookControl CoNtRoL;
-	private Scanner input;
-	private UI_STATE StATe;
+	private ReturnBookControl returnBookUiControl;
+	private Scanner keybordInput;
```
Comment (MinaTamang) [inline ReturnBookUI.java:9]: can you change keybordInput into keyboardInput

File: BorrowBookControl.java [hunk 2]
```diff
@@ -2,116 +2,111 @@
...
+	private library library1;
+	private member member;
+
+	private enum BookControlState {
+		INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED
+	};
+
+	private BookControlState borrowBookState;
+
+	private List<book> pendingList;
```
Comment (MinaTamang) [inline BorrowBookControl.java:16]: @GoJoeyGo can you change private List<book> and private list<loan> into private List<Book> and <Loan>

File: BorrowBookControl.java [hunk 3]
```diff
@@ -2,116 +2,111 @@
...
+			pendingList = new ArrayList<>();
+			borrowBookUi.setBorrowBookUiState(BorrowBookUI.BorrowBookUiState.SCANNING);
+			borrowBookState = BookControlState.SCANNING;
+		} else {
+			borrowBookUi.Display("Member cannot borrow at this time");
+			borrowBookUi.setBorrowBookUiState(BorrowBookUI.BorrowBookUiState.RESTRICTED);
+		}
+	}
+
+	public void scanneBook(int bookId) {
```
Comment (MinaTamang) [inline BorrowBookControl.java:53]: @GoJoeyGo scanneBook into scanBook just a simple spelling mistake.
Thanks

File: BorrowBookUI.java [hunk 4]
```diff
@@ -1,117 +1,98 @@
...
-	public static enum UI_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+	public static enum BorrowBookUiState {
+		INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED
+	};
 
-	private BorrowBookControl CONTROL;
-	private Scanner input;
-	private UI_STATE StaTe;
+	private BorrowBookControl control;
+	private Scanner keybordinput;
```
Comment (MinaTamang) [inline BorrowBookUI.java:9]: @GoJoeyGo keybordinput into keyboardInput. Please

File: BorrowBookUI.java [hunk 5]
```diff
@@ -1,117 +1,98 @@
...
 				output("Unhandled state");
-				throw new RuntimeException("BorrowBookUI : unhandled state :" + StaTe);			
+				throw new RuntimeException("BorrowBookUI : unhandled state :" + uiState);
 			}
-		}		
+		}
 	}
 
-
 	public void Display(Object object) {
```
Comment (MinaTamang) [inline BorrowBookUI.java:95]: @GoJoeyGo Display into display please

File: FixBookControl.java [hunk 6]
```diff
@@ -1,72 +1,62 @@
...
 	}
 
-
-	public void FIX_Book(boolean MUST_fix) {
-		if (!StAtE.equals(CONTROL_STATE.FIXING)) {
-			throw new RuntimeException("FixBookControl: cannot call fixBook except in FIXING state");
-		}	
-		if (MUST_fix) {
-			LIB.Repair_BOOK(Cur_Book);
+	public void setFixBook(boolean bookFixState) {
```
Comment (MinaTamang) [inline FixBookControl.java:44]: @GoJoeyGo boolean bookFixState into isBookFixed. but I am still not sure can you check in this one as boolean is used to know the true or false so.
Thanks

File: FixBookUI.java [hunk 7]
```diff
@@ -1,88 +1,70 @@
...
+		INITIALISED, READY, FIXING, COMPLETED
+	};
 
-	public static enum UI_STATE { INITIALISED, READY, FIXING, COMPLETED };
-
-	private FixBookControl CoNtRoL;
-	private Scanner input;
-	private UI_STATE StAtE;
+	private FixBookControl fixBookControl;
+	private Scanner keybordInput;
```
Comment (MinaTamang) [inline FixBookUI.java:9]: @GoJoeyGo keybordInput into keyboardInput. Thankyou

File: Main.java [hunk 8]
```diff
@@ -1,222 +1,211 @@
...
-		  .append("  P  : pay fine\n")
-		  .append("\n")
-		  .append("  T  : increment date\n")
-		  .append("  Q  : quit\n")
-		  .append("\n")
-		  .append("Choice : ");
-		  
-		return sb.toString();
+
+	private static Scanner keybordInput;
```
Comment (MinaTamang) [inline Main.java:6]: @GoJoeyGo keybordInput into keyboardInput please.

File: PayFineControl.java [hunk 9]
```diff
@@ -1,65 +1,59 @@
...
-		StAtE = CONTROL_STATE.INITIALISED;
+		this.library = library.instanceLibrary();
+		fineState = FineControlState.INITIALISED;
 	}
 	
-	
-	public void Set_UI(PayFineUI ui) {
-		if (!StAtE.equals(CONTROL_STATE.INITIALISED)) {
-			throw new RuntimeException("PayFineControl: cannot call setUI except in INITIALISED state");
+	public void payFineUi(PayFineUI ui) {
```
Comment (MinaTamang) [inline PayFineControl.java:15]: @GoJoeyGo  payFineUi into setPayFineUi if thats seems okay. Thankyou

File: ReturnBookControl.java [hunk 10]
```diff
@@ -1,76 +1,76 @@
 public class ReturnBookControl {
 
-	private ReturnBookUI Ui;
-	private enum CONTROL_STATE { INITIALISED, READY, INSPECTING };
-	private CONTROL_STATE sTaTe;
+	private ReturnBookUI returnBookUi;
+	private enum RetunrBookControlState { INITIALISED, READY, INSPECTING };
```
Comment (MinaTamang) [inline ReturnBookControl.java:4]: RetunrBookControlState into ReturnBookControlState just a spelling mistake @GoJoeyGo thankyou

File: ReturnBookControl.java [hunk 11]
```diff
@@ -1,76 +1,76 @@
...
-		this.lIbRaRy = lIbRaRy.INSTANCE();
-		sTaTe = CONTROL_STATE.INITIALISED;
+		this.library = library.instanceLibrary();
+		state = RetunrBookControlState.INITIALISED;
 	}
 	
 	
-	public void Set_UI(ReturnBookUI ui) {
-		if (!sTaTe.equals(CONTROL_STATE.INITIALISED)) {
+	public void returnBookSetUi(ReturnBookUI ui) {
```
Comment (MinaTamang) [inline ReturnBookControl.java:17]: @GoJoeyGo I think setReturnBookUi seems little good then returnBookSetUi. can you fix it. Thanks

**Review by MinaTamang — COMMENTED** [review]
@GoJoeyGo can you please change line 9 private scanner keybordinput into keyboardInput aand line 112 public void display into Display().
Thanks

File: BorrowBookControl.java (hunk 2 above)
Comment (GoJoeyGo) [reply BorrowBookControl.java:16]: what do you mean

File: BorrowBookControl.java (hunk 3 above)
Comment (GoJoeyGo) [reply BorrowBookControl.java:53]: happy to change thank you for noticing

File: BorrowBookUI.java (hunk 4 above)
Comment (GoJoeyGo) [reply BorrowBookUI.java:9]: happy to change thank you for noticing

File: BorrowBookUI.java (hunk 5 above)
Comment (GoJoeyGo) [reply BorrowBookUI.java:95]: display

File: FixBookUI.java (hunk 7 above)
Comment (GoJoeyGo) [reply FixBookUI.java:9]: will change keybordInput to keyboardInput i think i did it for all of the code

File: Main.java (hunk 8 above)
Comment (GoJoeyGo) [reply Main.java:6]: i just can spell keyboard will change

File: PayFineControl.java (hunk 9 above)
Comment (GoJoeyGo) [reply PayFineControl.java:15]: payFineUi can be changed to setPayFineUi

File: ReturnBookControl.java (hunk 10 above)
Comment (GoJoeyGo) [reply ReturnBookControl.java:4]: happy to change

File: ReturnBookControl.java (hunk 11 above)
Comment (GoJoeyGo) [reply ReturnBookControl.java:17]: ReturnBookUi will be changed to setReturnBookUi

File: ReturnBookUI.java (hunk 1 above)
Comment (GoJoeyGo) [reply ReturnBookUI.java:9]: yep

**Review by GoJoeyGo — COMMENTED** [review]
i approve all of the recommend changes and sugestions you have given me mina gits not letting me approve them so i am writing a comment to say that i have reviewed and looked though them all

File: BorrowBookControl.java (hunk 2 above)
Comment (MinaTamang) [reply BorrowBookControl.java:16]: @GoJoeyGo <book> into <Book> and <loan> into <Loan>

Comment (MinaTamang) [reply BorrowBookControl.java:16]: <Book> into <book>

Comment (MinaTamang) [reply BorrowBookControl.java:16]: book into <Book> please

File: ReturnBookUI.java (hunk 1 above)
Comment (MinaTamang) [reply ReturnBookUI.java:9]: cool

File: BorrowBookControl.java (hunk 3 above)
Comment (MinaTamang) [reply BorrowBookControl.java:53]: no worries @GoJoeyGo

File: BorrowBookUI.java (hunk 5 above)
Comment (MinaTamang) [reply BorrowBookUI.java:95]: yes

File: FixBookUI.java (hunk 7 above)
Comment (MinaTamang) [reply FixBookUI.java:9]: heheh yes but not a big deal

File: PayFineControl.java (hunk 9 above)
Comment (MinaTamang) [reply PayFineControl.java:15]: great

File: ReturnBookControl.java (hunk 10 above)
Comment (MinaTamang) [reply ReturnBookControl.java:4]: thanks

File: ReturnBookControl.java (hunk 11 above)
Comment (MinaTamang) [reply ReturnBookControl.java:17]: Great

**Comment by MinaTamang** [comment]
> i approve all of the recommend changes and sugestions you have given me mina gits not letting me approve them so i am writing a comment to say that i have reviewed and looked though them all

Okay Joey thanks

**Review by mpalme24 — CHANGES_REQUESTED** [review]
Line #86 of BorrowBookControl -  you have borrowBookUi.setBorrowBookUiState -but are then using BorrowBookUI in the brackets. Is this attended? 

Line # 10 of BorrowBookUI - you have "keybordinput" - can this be changed to "keyboardInput"? this is also on line #14 and #21 

Line #66 of BorrowBookUI has "control.scanneBook" - can this be changed to "control.scannedBook"?

Calender.java = fine 

Line #2 of FixBookControl - can fixBookUi be changed to fixBookUI to follow code standards 

This will also need to be changed on line 21, 22, 32, 36, 39

**Comment by MinaTamang** [comment]
> > On line #15 and #11 - you have "BookState", could it be more appropriate for it to be bookState?
> 
> Additionally we have
> private String bookTitle;
> private String bookAuthor;
> private String bookCallNumber;
> 
> So changing BookState to bookState would follow the rest of the codes guidelines.

@mpalme24 we can't change BookState to bookState because it is enum. Thankyou

**Comment by MinaTamang** [comment]
> Line #86 of BorrowBookControl - you have borrowBookUi.setBorrowBookUiState -but are then using BorrowBookUI in the brackets. Is this attended?
> 
> Line # 10 of BorrowBookUI - you have "keybordinput" - can this be changed to "keyboardInput"? this is also on line #14 and #21
> 
> Line #66 of BorrowBookUI has "control.scanneBook" - can this be changed to "control.scannedBook"?
> 
> Calender.java = fine
> 
> Line #2 of FixBookControl - can fixBookUi be changed to fixBookUI to follow code standards
> 
> This will also need to be changed on line 21, 22, 32, 36, 39

@mpalme24 Its done but I didn't change FixBookUi into FixBookUI sorry. Thanks

**Pull request merged**